### PR TITLE
[JENKINS-64338] - mask credentials also outside of log

### PR DESCRIPTION
**Pull request by scddev**
https://issues.jenkins.io/browse/JENKINS-64338

To have proper masking like in credential-plugin, a similar approach is chosen by using also a Step instead of the BuildWrapper.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+            String prefixPath = StringUtils.isBlank(config.getPrefixPath())
+                ? ""
+                : Util.ensureEndsWith(envVars.expand(config.getPrefixPath()), "/");
+
+            VaultAccessor vaultAccessor = new VaultAccessor();
+            vaultAccessor.setConfig(vaultConfig);
+            vaultAccessor.setCredential(credential);
+            vaultAccessor.setMaxRetries(config.getMaxRetries());
+            vaultAccessor.setRetryIntervalMilliseconds(config.getRetryIntervalMilliseconds());
+            vaultAccessor.init();
```
Comment (jetersen) [inline src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java:152]: This feel like duplicated 3 times 🤔

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 2]
```diff
@@ -261,7 +260,6 @@ public ConsoleLogFilter createLoggerDecorator(
      * that it can be accessed from views.
      */
     @Extension
-    @Symbol("withVault")
```
Comment (jetersen) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java:264]: nice! Thanks for getting rid of the hack.
This seems to resolve #51 ?

**Review by jetersen — COMMENTED** [review]
Thanks for the contribution.

File: src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java (hunk 1 above)
Comment (scddev) [reply src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java]: Sadly yes, I am not sure if I should start a refactoring to get rid of the duplication with a fix or this should be handled in a separate "cleanup" issue.

Comment (jetersen) [reply src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java]: Feel free to include the refactoring in this PR as the duplication is getting somewhat out of hand in this plugin already 😓

**Comment by jetersen** [comment]
Seems like the test are broken on Jenkins 2.277.1, JDK 8 and ubuntu-20.04

**Comment by jetersen** [comment]
Neat, I forgot that we actually do have non mock test for withVault
https://github.com/jenkinsci/hashicorp-vault-plugin/blob/2e2760b322ee470e638437187616f46b0a89fadf/src/test/resources/com/datapipe/jenkins/vault/util/pipeline.groovy#L12

https://github.com/jenkinsci/hashicorp-vault-plugin/blob/2e2760b322ee470e638437187616f46b0a89fadf/src/test/java/com/datapipe/jenkins/vault/jcasc/secrets/VaultSecretSourceTest.java#L280-L292

File: src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java (hunk 1 above)
Comment (scddev) [reply src/main/java/com/datapipe/jenkins/vault/VaultBindingStep.java]: Moved common code into VaultAccessor for now

**Review by jetersen — APPROVED** [review]
LGTM, is there anything missing on your end @scddev ?

**Comment by scddev** [comment]
Looks fine for me also.

**Pull request merged**